drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,959 @@
+
+};
+
+static const struct T_BaudrateStruct mBaudrateTb[t_br_max_idx] = {
+/**                       1,500,000
+ *              NUM      (3,000,000)         1
+ * ( (INT+1) + ----- ) = ----------- x --------------
+ *              DEN       baud rate    ((DLM<<8)|DLL)
+ *
+ *    INT, DEN, NUM,  DLL,  DLM
+ */
+	{ 155, 100,  25,   0x18, 0x00 },        // 0x00 - 4800
+	{ 155, 100,  25,   0x10, 0x00 },        // 0x01 - 9600
+	{ 103,   6,   1,   0x0c, 0x00 },        // 0x02 - 14400
+	{  77,   8,   1,   0x06, 0x00 },        // 0x03 - 19200
+	{  38,  16,   1,   0x03, 0x00 },        // 0x04 - 38400
+	{  25,  24,   1,   0x02, 0x00 },        // 0x05 - 57600
+	{  12,  48,   1,   0x01, 0x00 },        // 0x06 - 115200
+	{   5,  96,  49,   0x02, 0x80 },        // 0x07 - 230400
+	{   2, 192,  49,   0x01, 0x80 },        // 0x08 - 460800
+};
+
+/**
+ *
+ * uart_putc
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static uint8_t uart_putc(uint8_t chan, uint8_t data)
+{
+	unsigned long u32temp = 0;
+
+  #if DEF_ENABLE_CTS_RTS
+	uint8_t Status  = mComPortTb[chan].mUartBase->UMSR;
+  #endif /* DEF_ENABLE_CTS_RTS */
+
+	while ((mComPortTb[chan].mUartBase->ULSR & LSR_THRE) == 0) {
+		if (u32temp == 0x100) {
+			return FALSE;
+		}
+		u32temp++;
+	}
+
+	mComPortTb[chan].mUartBase->UTHR = data;
+	return TRUE;
+}
+
+/**
+ *
+ * uart_getc
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static uint8_t uart_getc(uint8_t chan, uint8_t *data)
+{
+	unsigned long u32temp = 0;
+
+  #if DEF_ENABLE_CTS_RTS
+	uint8_t Status  = mComPortTb[chan].mUartBase->UMSR;
+  #endif /* DEF_ENABLE_CTS_RTS */
+
+	while ((mComPortTb[chan].mUartBase->ULSR & LSR_DR) == 0) {
+		if (u32temp == 0x100) {
+			return FALSE;
+		}
+		u32temp++;
+	}
+
+	*data = mComPortTb[chan].mUartBase->UTHR;
+	return TRUE;
+}
+
+/**
+ *
+ * uart_is_xmit_empty
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static uint8_t uart_is_xmit_empty(uint8_t chan)
+{
+	return IS_MASK_SET(mComPortTb[chan].mUartBase->ULSR, LSR_TEMT);
+}
+/**
+ *
+ * uart_init
+ *
+ * @return
+ *
+ * @parameter
+ *
+ *
+ */
+static void uart_init(uint8_t chan, uint8_t baudrate, uint8_t bINTMode,
+		      uint8_t bIsEnableParity,
+		      uint8_t bIsEvenParity,
+		      uint8_t bIsOneStopBit)
+{
+	volatile struct T_NS16550  *pComPort
+		= mComPortTb[chan].mUartBase;
+	const struct T_BaudrateStruct *pBR
+		= &mBaudrateTb[baudrate];
+	unsigned char ucTemp;
+
+	/* Enable auto gating */
+	/* UART Clock Gating (UART12CG) 0: Operation,
+	 * 1: Clocks to UART1/UART2 modules are gated.
+	 */
+	CGCTRL3R &= ~BIT2;
+	AUTOCG   &= ~(BIT(6 - chan));
+
+	/* Enable uart module */
+	RSTDMMC  |= BIT(3 - chan);      /* Set EC side control */
+	RSTC4    = BIT(1 + chan);       /* W-One to reset controller */
+	GCR1    |= mComPortTb[chan].ucGCR1;
+	GCR6    |= mComPortTb[chan].ucGCR6;
+	GCR8    |= mComPortTb[chan].ucGCR8;
+
+	if (chan == t_uart_1) {
+		GCR21 &= ~(GPH1VS | GPH2VS);
+		SET_MASK(UART2_ECSPMR, BIT1);
+		#if DEF_UART2_ENABLE_CTS_RTS    // {
```

Comment:
This should be horizontally aligned with the #endif on line 372 below. I would put both in column 1.

Reply:
Also is there a reason for the "// {" at the end of the line? It looks like it will interfere with brace matching.